test/e2e/framework/util.go

```diff
@@ -4447,12 +4449,22 @@ const (
 func CheckConnectivityToHost(f *Framework, nodeName, podName, host string, pingCmd PingCommand, timeout int) error {
 	contName := fmt.Sprintf("%s-container", podName)
 
-	command := []string{
-		string(pingCmd),
-		"-c", "3", // send 3 pings
-		"-W", "2", // wait at most 2 seconds for a reply
-		"-w", strconv.Itoa(timeout),
-		host,
+	var command []string
+	if !ProviderIs("azure") {
+		command = []string{
+			string(pingCmd),
+			"-c", "3", // send 3 pings
+			"-W", "2", // wait at most 2 seconds for a reply
+			"-w", strconv.Itoa(timeout),
+			host,
+		}
+	} else {
+		command = []string{
+			string(pingCmd),
```

Comment:
This pattern seems really awkward to me. We're passing in a generic `PingCommand` type but we are then trying to pass in differing arguments into it.

Reply:
I encaped all implementation details in ```CheckConnectivityToHost``` now.